### PR TITLE
Remove NET452 NET46 from OpenTelemetry sdk

### PR DESCRIPTION
**Pull request by cijothomas**
Part of https://github.com/open-telemetry/opentelemetry-dotnet/issues/2138

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by utpilla — APPROVED** [review]

**Comment by cijothomas** [comment]
Waiting for one more approval, as this is bit larger PR, and it touches the heart of this repo!

File: src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs
Comment (reyang) [inline src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs:327]: Indent?

File: src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs
Comment (reyang) [inline src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs:59]: Finally 👍 

IIRC there are many places where we have `(EventKeywords)(-1)`, probably not fully covered by this PR. Can be a separate PR though.

File: src/OpenTelemetry/OpenTelemetry.csproj
Comment (reyang) [inline src/OpenTelemetry/OpenTelemetry.csproj:3]: After this PR we might be able to consolidate these `TargetFrameworks` (e.g. using the default setting in the top level `.props` file).

**Review by reyang — APPROVED** [review]
LGTM.

**Pull request merged**
